components/script/script_thread.rs

```diff
@@ -1751,6 +1759,23 @@ impl ScriptThread {
 
         // Receive at least one message so we don't spinloop.
         debug!("Waiting for event.");
+        let mut event = select! {
+            recv(self.task_queue.select()) -> msg => {
+                self.task_queue.take_tasks(msg.unwrap());
+                let event = self
+                    .task_queue
+                    .recv()
+                    .expect("Spurious wake-up of the event-loop, task-queue has no tasks available");
+                FromScript(event)
+            },
+            recv(self.control_port) -> msg => FromConstellation(msg.unwrap()),
+            recv(self.devtools_chan.as_ref().map(|_| &self.devtools_port).unwrap_or(&crossbeam_channel::never())) -> msg
+                => FromDevtools(msg.unwrap()),
+            recv(self.image_cache_port) -> msg => FromImageCache(msg.unwrap()),
+        };
+
+        // select! macro parses all arms regardless of #[cfg], we need to add a separate block for WebGPU
+        #[cfg(feature = "webgpu")]
```

Comment:
this is bad, can we fix select macro?

Reply:
I was suppose to revisit this, I forgot! thanks for noticing! I have applied cfg in [91c6cdf](https://github.com/servo/servo/pull/34444/commits/91c6cdf34487abb9e55cd38d168e5c222d9c09b0)